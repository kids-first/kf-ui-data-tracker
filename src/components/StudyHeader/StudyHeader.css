.StudyHeader--TopText {
  @extend .row-1, .cell-12, .font-body, .text-darkGrey;
}

.StudyHeader--TitleText {
  @extend .block, .text-blue, .leading-none, .font-light, .text-4xl;
}

.StudyHeader--TopText-loading {
  @extend .row-1, .cell-4, .text-white, .bg-white;
}

.StudyHeader--TitleText-loading {
  @extend .pt-16, .pb-20, .mr-16, .bg-white;
}

.StudyHeader--InfoText-loading {
  @extend .invisible, .text-white, .bg-white;
}
