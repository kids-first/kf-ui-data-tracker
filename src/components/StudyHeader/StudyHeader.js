import React from 'react';
import propTypes from 'prop-types';
import CopyButton from '../CopyButton/CopyButton';
import {Container, Header, Placeholder} from 'semantic-ui-react';

const StudyHeader = ({
  kfId,
  modifiedAt,
  shortName,
  name: studyName,
  loading,
}) => {
  if (loading) {
    return (
      <Container>
        <Placeholder>
          <Placeholder.Header>
            <Placeholder.Line />
            <Placeholder.Line />
          </Placeholder.Header>
        </Placeholder>
      </Container>
    );
  }

  return (
    <Container>
      <Header as="h1">{studyName}</Header>
      <Header.Subheader as="h2" className="study--header-sub">
        {shortName}
      </Header.Subheader>
      {kfId && <CopyButton basic text={kfId} size="tiny" />}
    </Container>
  );
};

StudyHeader.propTypes = {
  /** the kf_id (SD_XXXXXXX) for the study  */
  kfId: propTypes.string,
  /** datetime of when the study was last updated */
  modifiedAt: propTypes.string,
  /** user friendly name for the study  */
  shortName: propTypes.string,
  /** long programatic name for the study  */
  name: propTypes.string,
  /** Loading state of the study header*/
  loading: propTypes.bool,
};

export default StudyHeader;
