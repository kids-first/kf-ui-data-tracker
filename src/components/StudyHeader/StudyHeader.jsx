import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';
import TimeAgo from 'react-timeago';
import CopyButton from '../CopyButton/CopyButton';

const StudyHeader = ({
  kfId,
  modifiedAt,
  shortName,
  name: studyName,
  className,
}) => {
  const studyHeaderClasses = classes('StudyHeader', 'BodyContent', className);
  return (
    <div className="bg-lightGrey">
      <header className={studyHeaderClasses}>
        <div>
          <h3 className="mt-1 text-base">
            <span className="font-body font-bold pr-4 text-darkGrey" id="kfId">
              {kfId}
            </span>
            <CopyButton text={kfId} />
            <span className="study-modified-date pl-20 font-light">
              <small>
                last updated: <TimeAgo date={modifiedAt} />
              </small>
            </span>
          </h3>
          <h2 className="mt-4 font-light text-blue text-4xl">
            {shortName || studyName}
          </h2>
        </div>
        <div>
          <h4 className="mt-2 text-sm font-bold">Contacts:</h4>
          <p className="mt-1 font-light">Mary Marazita</p>
        </div>
      </header>
    </div>
  );
};

StudyHeader.propTypes = {
  /** the kf_id (SD_XXXXXXX) for the study  */
  kfId: propTypes.string.isRequired,
  /** datetime of when the study was last updated */
  modifiedAt: propTypes.string.isRequired,
  /** user friendly name for the study  */
  shortName: propTypes.string,
  /** long programatic name for the study  */
  name: propTypes.string,
  /** Any additional classes to be applied to the button */
  className: propTypes.string,
};

export default StudyHeader;
