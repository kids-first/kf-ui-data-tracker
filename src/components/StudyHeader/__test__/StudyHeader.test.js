import React from 'react';
import {render} from 'react-testing-library';
import StudyHeader from '../StudyHeader';

it('renders study header with study name and short name', () => {
  const tree = render(
    <StudyHeader
      kfId="SD_00000000"
      shortName="Test Study"
      name="A study used for testing the header"
      modifiedAt={'2019-01-01'}
    />,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders study header in loading stage', () => {
  const tree = render(<StudyHeader loading />);
  expect(tree.container).toMatchSnapshot();
});

it('renders study header with no data', () => {
  const tree = render(<StudyHeader />);
  expect(tree.container).toMatchSnapshot();
});

it('renders study header with different study creation status -- creating', () => {
  const tree = render(
    <StudyHeader
      kfId="SD_00000000"
      shortName="Test Study"
      name="A study used for testing the header"
      modifiedAt={'2019-01-01'}
      newStudy={['SD_00000000_STR']}
    />,
  );
  expect(tree.container).toMatchSnapshot();
  expect(
    tree.queryByText(
      /External study resources creation in progress. Click for details./i,
    ),
  ).not.toBeNull();
});

it('renders study header with different study creation status -- err', () => {
  const tree = render(
    <StudyHeader
      kfId="SD_00000000"
      shortName="Test Study"
      name="A study used for testing the header"
      modifiedAt={'2019-01-01'}
      newStudy={['SD_00000000_ERR']}
    />,
  );
  expect(tree.container).toMatchSnapshot();
  expect(
    tree.queryByText(
      'Error(s) creating external study resources. Click for details.',
    ),
  ).not.toBeNull();
});

it('renders study header with different study creation status -- created', () => {
  const tree = render(
    <StudyHeader
      kfId="SD_00000000"
      shortName="Test Study"
      name="A study used for testing the header"
      modifiedAt={'2019-01-01'}
      newStudy={['SD_00000000_SUC']}
    />,
  );
  expect(tree.container).toMatchSnapshot();
  expect(
    tree.queryByText(/External study resources created! Click for details./i),
  ).not.toBeNull();
});
