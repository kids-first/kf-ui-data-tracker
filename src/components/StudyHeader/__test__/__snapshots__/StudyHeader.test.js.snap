// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders study header in loading stage 1`] = `
<div>
  <div
    class="ui container"
  >
    <div
      class="ui placeholder"
    >
      <div
        class="header"
      >
        <div
          class="line"
        />
        <div
          class="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders study header with different study creation status -- created 1`] = `
<div>
  <div
    class="ui container"
  >
    <small
      class="text-green cursor-pointer noMargin"
    >
      <i
        aria-hidden="true"
        class="check circle icon"
      />
      <span
        class="text-underline"
      >
        External study resources created! Click for details.
      </span>
    </small>
    <h1
      class="ui header mt-15"
    >
      A study used for testing the header
    </h1>
    <h2
      class="sub header study--header-sub"
    >
      Test Study
    </h2>
    <button
      class="ui tiny basic icon button"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
      SD_00000000
    </button>
  </div>
</div>
`;

exports[`renders study header with different study creation status -- creating 1`] = `
<div>
  <div
    class="ui container"
  >
    <small
      class="text-blue cursor-pointer noMargin"
    >
      <i
        aria-hidden="true"
        class="clock icon"
      />
      <span
        class="text-underline"
      >
        External study resources creation in progress. Click for details.
      </span>
    </small>
    <h1
      class="ui header mt-15"
    >
      A study used for testing the header
    </h1>
    <h2
      class="sub header study--header-sub"
    >
      Test Study
    </h2>
    <button
      class="ui tiny basic icon button"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
      SD_00000000
    </button>
  </div>
</div>
`;

exports[`renders study header with different study creation status -- err 1`] = `
<div>
  <div
    class="ui container"
  >
    <small
      class="text-red cursor-pointer noMargin"
    >
      <i
        aria-hidden="true"
        class="warning sign icon"
      />
      <span
        class="text-underline"
      >
        Error(s) creating external study resources. Click for details.
      </span>
    </small>
    <h1
      class="ui header mt-15"
    >
      A study used for testing the header
    </h1>
    <h2
      class="sub header study--header-sub"
    >
      Test Study
    </h2>
    <button
      class="ui tiny basic icon button"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
      SD_00000000
    </button>
  </div>
</div>
`;

exports[`renders study header with no data 1`] = `
<div>
  <div
    class="ui container"
  >
    <h1
      class="ui header"
    >
      Unknown study name
    </h1>
  </div>
</div>
`;

exports[`renders study header with study name and short name 1`] = `
<div>
  <div
    class="ui container"
  >
    <h1
      class="ui header"
    >
      A study used for testing the header
    </h1>
    <h2
      class="sub header study--header-sub"
    >
      Test Study
    </h2>
    <button
      class="ui tiny basic icon button"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
      SD_00000000
    </button>
  </div>
</div>
`;
